frontend/client/src/screens/CodeValidations.js

```diff
@@ -79,36 +85,42 @@ const CodeValidationsBase = observer((props) => {
     updateButtonDisabled()
   }
 
-  const handleDate = (e) => {
-    e.target.classList.add('with-value')
-    e.target.classList.remove('no-value')
+  const handleDate = (date) => {
+    setSymptomDateObject(date)
+    const selectedDate = toDashSeperatedYYYYMMDDString(date)
+    document.getElementById('date-picker').classList.add('with-value')
+    document.getElementById('date-picker').classList.remove('no-value')
 
     // does not allow symptomDate in state to be set if date selected is more than 14 days ago or in the future
-    if (moreThanFourteenDaysAgo(e.target.value)) {
+    if (moreThanFourteenDaysAgo(selectedDate)) {
```

Comment:
The `DatePicker` object does not allow a date outside its range to be selected. So no date passed to this function will be invalid. So we can delete these lines (95-109) and always run what is currently the else case. I think we can also delete the `dateInFuture` and `moreThanFourteenDaysAgo` functions from `time.js`, which is also good. 

In this recording: I put `console.log(date)` at Line 89. Note how for invalid dates, they are immediately reset to a valid date and not logged.  Also note the uncaught error when the date is cleared. 
![vals1](https://user-images.githubusercontent.com/20446774/90298440-24394a00-de47-11ea-8f89-e536c637f75f.gif)

Reply:
Yup!  I was just about to commit this deletion myself!